models/classes/pack/encoders/Base64fileEncoder.php

```diff
@@ -71,7 +71,7 @@ public function encode($data)
             $mediaSource = $data->getMediaSource();
             $data = $data->getMediaIdentifier();
 
-            if ($mediaSource instanceof HttpSource) {
+            if ($mediaSource instanceof HttpSource || preg_match('/^(data:.*;base64)/', $data)) {
```

Comment:
On the scope of the taoBooklet generation, during the pack-test phase we have the itemPack
`\oat\taoQtiItem\model\pack\QtiItemPacker::packQtiItem`  Where we use the ItemMediaResolver to resolve assets in items, here https://github.com/oat-sa/extension-tao-item/blob/master/models/classes/media/ItemMediaResolver.php#L45 .

I would prefer to tackle this case on the root. Resolve the asset as base64encoded and not use the default `LocalItemSource`.  Create new class something like `RawBase64EncodedSource` and catch it here and return accordingly.
wdyt?